substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/substitute/UnsafeAutomaticSubstitutionProcessor.java

```diff
@@ -419,6 +428,11 @@ private void processUnsafeObjectFieldOffsetInvoke(
          */
         ResolvedJavaField offsetField = extractValueStoreField(unsafeObjectFieldOffsetInvoke.asNode(), FieldOffset, unsuccessfulReasons);
 
+        // No field, but doesn't escape local usage, ignore.
+        if (offsetField == null && unsuccessfulReasons.isEmpty()) {
```

Comment:
I don't like the idea of encoding the 'does-not-escape' decision in `unsuccessfulReasons.isEmpty()`. That may not always be true. The way the analysis is currently structured first the right-hand-side of an unsafe call is processed, and failure reasons recorded, then the left-hand-side. Therefore, the second part of the analysis that tries to understand where the unsafe value is stored may already have some failure reasons from the right-hand-side, e.g., non-constant arguments to the unsafe call. We need the right-hand-side processed first to extract bits of information needed by the recomputation itself such as target field name and holder class. So, a better solution is to wrap the result in a class: 
``` 
    static final class SearchResult {
        /** The field where the value is stored, if found. */
        final ResolvedJavaField valueStoreField;
        /**
         * Uses that can lead to the unsafe value having side effects that we didn't account for are
         * considered illegal.
         */
        final boolean illegalUseFound;
}
```
In this paradigm the "doesn't escape local usage" is a legal use and therefore ignored.